x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -393,6 +405,17 @@ export class TaskManagerRunner implements TaskRunner {
     return { ...(error ? { error } : {}), state };
   }
 
+  private validateTaskState(taskInstance: ConcreteTaskInstance) {
```

Comment:
It looks like throwing this error here prevents the task from being updated, so the task status remains as `running` and will only get claimed again when the `retryAt` timestamp has passed (typically 5 minutes). Is this acceptable behavior? cc @mikecote

Reply:
@ymao1 @doakalexi Ideally in this scenario we should allow the task update request to succeed. However, I experimented with the code to see what changes would be required to allow this through and I can't see a good result without some code refactoring. Given this PR does solve a significant portion of the issue that would prevent other tasks from running, I think we can create a follow up issue to refactor / solve this scenario separately given it doesn't seem to be as straightforward. Wdyt?